packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveCmp.ts

```diff
@@ -111,92 +97,20 @@ export async function forceSourceRetrieveCmp(componentNode: BrowserNode) {
       : componentName;
   const fileExts = generateSuffix(typeNode, typeName);
 
-  const sourcePathStrategy = BUNDLE_TYPES.has(typeName)
-    ? new BundlePathStrategy()
-    : new DefaultPathStrategy();
+  const globStrategy = BUNDLE_TYPES.has(typeName)
+    ? GlobStrategyFactory.createCheckBundleInAllPackages(...fileExts)
+    : GlobStrategyFactory.createCheckFileInAllPackages(...fileExts);
 
   const executor = new ForceSourceRetrieveExecutor(typeName, componentName);
 
   const commandlet = new SfdxCommandlet(
-    workspaceChecker,
-    parameterGatherer,
+    new SfdxWorkspaceChecker(),
+    new DirFileNameGatherer(join('main', 'default', dirName), componentName),
```

Comment:
Very cool 👍

Reply:
The one thing I'd like to know is if we can/should avoid hardcoding `main/default` to the retrieve since I've seen people set up their projects with the following structure:

```
  - force-app
      - classes
      - triggers
```

instead of:

```
  -force-app
    - main
       - default
          - classes
          - triggers
```

Reply:
This hard coding used to be [here](https://github.com/forcedotcom/salesforcedx-vscode/pull/1629/files#diff-018c50cc4869276ca6a53b2421852471L191). The reason we're doing so at least for the retrieve component command is because the org browser calls `force:source:retrieve` with mostly default parameters, which will always retrieve to `main -> default -> somefolder`.